app/assets/main/react_components/routes/lifestyle_footprints/footprint_form_container/components/footprint_form/components/sign_up_page/components/membership_page/components/ReferralCode.jsx

```diff
@@ -9,7 +9,13 @@ import AnswerButton from '../../../../../components/common/AnswerButton.jsx';
 const ReferralCode = ({ grantedReferralCode, setGrantedReferralCode }) => {
   const [invalidCodeMessage, setInvalidCodeMessage] = useState('');
   const [inputCode, setInputCode] = useState(grantedReferralCode);
-
+  const [showInput, setShowInput] = useState(() => {
+    if (grantedReferralCode === '') {
+      return false;
+    } else {
+      return true;
+    }
+  });
```

Comment:
Skulle detta kunna förkortas med detta tror du? :) 
```suggestion
  const [showInput, setShowInput] = useState(grantedReferralCode === '' ? false : true);
```

Reply:
Eller hmm! Till och med detta:

```suggestion
  const [showInput, setShowInput] = useState(grantedReferralCode !== '');
```

Reply:
Då evalueras uttrycket bara rakt av, alltså stämmer grantedReferralCode !== '' ? I så fall true, annars false. Precis som man kan göra i metoders return-statement

Reply:
Coolt, visste inte att det gick att göra det så kort. Snyggt 👍

Reply:
Nej förstår det 😃  kul grej att lära sig tänkte jag!